fem/pgridfunc.cpp

```diff
@@ -225,10 +225,9 @@ void ParGridFunction::ExchangeFaceNbrData()
    MPI_Request *recv_requests = requests + num_face_nbrs;
    MPI_Status  *statuses = new MPI_Status[num_face_nbrs];
 
-   const double *h_data = this->HostRead();
    for (int i = 0; i < send_data.Size(); i++)
    {
-      send_data[i] = h_data[send_ldof[i]];
+      send_data[i] = data[send_ldof[i]];
```

Comment:
I think this change is wrong -- we want this method to work when `data` is in GPU memory too.

What is your argument for the change?

Reply:
Sorry about this.... I mixed up files. You are right, the update should be with exchanging face nbr data a few lines below. I've just pushed the correct fix.